### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by katarighe**
In this phase, I have completed the following tasks:
1) **Added** Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).

2) **Created** a directory that will contain all your Redux logic (`/src/redux`)

3) **Configure**d a Redux store (`/src/redux/store.js`)

4) **Defined** a slice of state for books that:
- Includes an array of books (initial state: empty array).
- Includes a reducer that adds a book.
- Includes a reducer that removes a book.

5) **Defined** a slice of state for categories that:
- Includes an array of categories (initial state: empty array).
- Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string).

6) **Structured** my application files using a "feature folder" approach and use the ducks pattern for its Redux files.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          author: action.payload.author,
+        },
+      ];
+
+    case 'REMOVE_BOOKS':
+      return state.filter((book) => book.id !== action.id);
+    default:
+      return state;
+  }
+};
```
Comment (LionRouge1) [inline src/redux/books/booksSlice.js]: - [ ] Nice work creating the reducer :clap: However it is requested in the project requirement to use `createSlice` API to define the state for book. It is the standard approach for writing Redux logic. You can read more about `createSlice` [here](https://redux-toolkit.js.org/api/createslice). Therefore I suggest you use it to follow the project requirement.
Screenshot from 2023-08-21 16-05-59

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const initalState = [];
+const categoriesReducer = (state = initalState, action) => {
+  switch (action.type) {
+    case CHECK_CATEGORY:
+      return ['Under construction'];
+    default:
+      return state;
+  }
+};
+
```
Comment (LionRouge1) [inline src/redux/categories/categoriesSlice.js]: - [ ] Same thing as pointed out earlier :point_up_2:

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @katarighe, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
-  Configure their Redux Store :heavy_check_mark: 
- Write actions and reducer for adding and removing books :heavy_check_mark: 
- All linters passed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @katarighe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitFlow`  ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 

## Optional suggestions 

**Please add the `Future Feature` section in your `README` file, because except FAQ rest of the everything is required which will help the viewer/user/developer to understand your project and vision of the project very well.**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**